File: leetcode1108.py
#Author: Zach Naymik
#Date: March 2, 2022

class Solution(object):
    def defangIPaddr(self, address):
        """
        :type address: str
        :rtype: str
        """
        new_str = ""
        for i in address:
            if i == ".":
                new_str += "[.]"
            else:
                new_str += i
        return new_str
        

File: leetcode1108.cc
//Author: Zach Naymik
//Date: March 5, 2022

#include <iostream>
#include <iomanip>
#include <string>
#include <cstdlib>
#include <cctype>

using namespace std;

string defangIPAddress(string address);

int main() {
     string address = "1.1.1.1";
     cout << defangIPAddress(address);
}

string defangIPAddress(string address) {
    string new_string;
    for (unsigned int i = 0; i < address.length(); i++) {
        if (address[i] == '.') {
            new_string += "[.]";
        }
        else {
            new_string += address[i];
        }
    }
    return new_string;
}

